cmd/paratime/denom/add.go

```diff
@@ -0,0 +1,49 @@
+package denom
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/spf13/cobra"
+
+	"github.com/oasisprotocol/oasis-sdk/client-sdk/go/config"
+
+	cliConfig "github.com/oasisprotocol/cli/config"
+)
+
+var addDenomCmd = &cobra.Command{
+	Use:   "add <network> <paratime> <denomination> <number_of_decimals>",
```

Comment:
How do you specify the native denomination?

Reply:
Currently you cant. We could do it like this:
`add <network> <paratime> <denomination> <number_of_decimals> --native`

or like this
`add <network> <paratime> <denomination-id> <denomination> <number_of_decimals>`

Which one do you prefer?

Reply:
You mean `<denomination>` and `<symbol>` or `--symbol <symbol>` (defaulting to `<denomination>` except for native where `--symbol` would be required)? And then you could use `_` or similar for native (in the config it needs to use the native denomination key).

Reply:
Alternative suggestion:
- `denom add-native <network> <paratime> <denomination> <number_of_decimals>` and
- `denom add-erc20 <network> <paratime> <smart_contract_addr>`

You can obtain the name, symbol and number of decimals from the contract automatically. I suppose we need to associate the ERC20 symbol with the actual contract anyway?

Reply:
This has nothing to do with ERc-20. These are SDK denominations.

Reply:
I do not have a strong opinion about this... Maybe would be best to do it like your second idea: 
`add <network> <paratime> <denomination> <number_of_decimals> --symbol <symbol>` (defaulting to `<denomination>` except for native where `--symbol` would be required). 

In this case user will have to know that `_` is for native. On the other hand this is quite advanced feature so more advanced user will find this info somewhere. Can I do it like this?

Reply:
> I do not have a strong opinion about this... Maybe would be best to do it like your second idea: `add <network> <paratime> <denomination> <number_of_decimals> --symbol <symbol>` (defaulting to `<denomination>` except for native where `--symbol` would be required).
> 
> In this case user will have to know that `_` is for native. On the other hand this is quite advanced feature so more advanced user will find this info somewhere. Can I do it like this?

The native denomination is already mandatory in the `paratime add` command. Maybe, it would be more appropriate to discriminate between the native/non-native behavior in the CLI too and introduce a new `oasis paratime set-native-denomination <network> <paratime> <symbol> <number_of_decimals>` command instead.

Reply:
^ i think it would be nice to keep the `denom` related cmds in `oasis paratime denom`. because later we will add also `remove`.

ok, what if we do:
- `oasis paratime denom add <network> <paratime> <denomination> <number_of_decimals> --symbol <symbol>` (if someone sets denomination to `_` it will not allow them to change denom)
- `oasis paratime denom set-native <network> <paratime> <denomination> <number_of_decimals>`

?

Reply:
Yeah, that could work. Also keep in mind that you shouldn't be able to remove the native denominator.